### PR TITLE
Optimize the code for testing sqlParser using MySQL Test Suite.

### PR DESCRIPTION
**Pull request by iamhucong**
Changes proposed in this pull request:
  - Adjusted some parsing rules for MySQL Test Suite .test files.

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/external/loader/ExternalMySQLTestParameterLoader.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    public Collection<ExternalSQLParserTestParameter> createTestParameters(final String sqlCaseFileName,
+                                                                           final List<String> sqlCaseFileContent,
+                                                                           final List<String> resultFileContent, final String databaseType, final String reportType) {
+        Collection<ExternalSQLParserTestParameter> result = new LinkedList<>();
+        List<String> sqlLines = new ArrayList<>();
+        int sqlCaseEnum = 1;
+        String delimiter = ";";
+        for (String line : sqlCaseFileContent) {
+            String trimLine = line.trim();
+            if (trimLine.isEmpty() || sqlLines.size() == 0 && isComment(trimLine)) {
```
Comment (strongduanmu) [inline test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/external/loader/ExternalMySQLTestParameterLoader.java:58]: Please put constant to left condition.

File: test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/external/loader/ExternalMySQLTestParameterLoader.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                sqlLines.clear();
+            }
+        }
+        return result;
+    }
+    
+    private String getNewDelimiter(final String delimiterSQL, final String delimiter) {
+        String trimSQL = delimiterSQL.trim();
+        String newDelimiter = trimSQL
+                .substring(9, trimSQL.endsWith(delimiter) ? trimSQL.length() - delimiter.length() : trimSQL.length())
```
Comment (strongduanmu) [inline test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/external/loader/ExternalMySQLTestParameterLoader.java:82]: Please do not use magic number.

File: test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/external/loader/ExternalMySQLTestParameterLoader.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        return newDelimiter.isEmpty() ? delimiter : newDelimiter;
+    }
+    
+    private boolean isComment(final String statement) {
+        return statement.startsWith("#") || statement.startsWith("/") || statement.startsWith("--") || statement.startsWith(":") || statement.startsWith("\\");
+    }
+    
+    private boolean existInResultContent(final List<String> resultLines, final List<String> sqlLines) {
+        int nextLineIndex = searchSQLNextLineIndex(resultLines, sqlLines);
+        return nextLineIndex != -1 && (nextLineIndex == resultLines.size() || !resultLines.get(nextLineIndex).contains("ERROR"));
```
Comment (strongduanmu) [inline test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/external/loader/ExternalMySQLTestParameterLoader.java:96]: Please put constant to left condition.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by iamhucong** [comment]
@strongduanmu Hi, already modified the code as per the suggestion.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**